compiler-core/src/type_/tests/warnings.rs

```diff
@@ -563,3 +563,113 @@ fn double_unary_bool_variable() {
         "#
     );
 }
+
+#[test]
+fn prefer_list_is_empty_over_list_length() {
```

Comment:
Let's use snapshots for these tests rather than comparing to the warning object. That way we can see what the rendered error looks like and test that part too.

Reply:
Good call, that makes this much nicer.

I no longer need to build the compiler and run it against a test Gleam project every time I want to visually inspect the warnings 😄